domains/pallets/auto-id/src/lib.rs

```diff
@@ -138,6 +146,33 @@ pub struct AutoId {
     pub certificate: Certificate,
 }
 
+impl AutoId {
+    /// Deterministically derives an identifier for the `AutoId`.
+    ///
+    /// The identifier is derived by hashing the subject common name of the certificate.
+    /// If the certificate is a leaf certificate, the issuer identifier is combined with the subject common name.
+    fn derive_identifier(&self) -> Identifier {
+        match &self.certificate {
+            Certificate::X509(cert) => {
+                let subject_common_name = &cert.subject.common_name.0[..];
+
+                if let Some(issuer_id) = cert.issuer_id {
+                    let mut data = [0u8; 32];
+                    issuer_id.to_big_endian(&mut data);
```

Comment:
> U256 is [u64; 4], so needs conversion to bytes.

It should not be U256 anymore. It should a `h256` since this is the AutoID identifier of the issuer. No difference from the AutoId of the leaf here